acceptance/tests/base/dsl/helpers/web_helpers/web_helpers_test.rb

```diff
@@ -0,0 +1,60 @@
+require 'helpers/test_helper'
+require 'webrick'
+require 'webrick/https'
+
+test_name 'dsl::helpers::web_helpers #link_exists?' do
+
+  http_cmd = "ruby -rwebrick -e'WEBrick::HTTPServer.new(:Port => 80, :DocumentRoot => \"/tmp\").start' > /tmp/mylogfile 2>&1 &"
+  master = only_host_with_role(hosts, 'master')
+  on(master, http_cmd)
+  dir = master.tmpdir('test_dir')
+  file = master.tmpfile('test_file')
+  dir.slice! "/tmp"
+  file.slice! "/tmp"
+  dst_dir = 'web_helpers'
+
+  step '#port_open_within? can tell if a port is open' do
+    assert port_open_within?(master,80,1)
+  end
+
+  step '#link_exists? can tell if a basic link exists' do
+    assert link_exists?("http://#{master}")
+  end
+
+  step '#link_exists? can tell if a basic link does not exist' do
+    assert !link_exists?("http://#{master}/test")
+  end
+
+  step '#link_exists? can use an ssl link' do
+
+    cert_name = [
+        %w[CN localhost],
+    ]
+
+    http_cmd = "ruby -rwebrick/https -e'WEBrick::HTTPServer.new(:SSLEnable => true, :SSLCertName => #{cert_name}, :Port => 4430,
+:DocumentRoot => \"/tmp\").start' > /tmp/mylogfile 2>&1 &"
+
+    sleep(2)
+    on(master, http_cmd)
+    assert link_exists?("https://#{master}:4430")
+  end
+
+  step '#fetch_http_dir can fetch a dir' do
+    assert_equal "#{dst_dir}#{dir}", fetch_http_dir("http://#{master}/#{dir}", dst_dir)
+  end
+
+  step '#fetch_http_dir will raise an error if unable fetch a dir' do
+    exception = assert_raises(RuntimeError) { fetch_http_dir("http://#{master}/tmps", dst_dir) }
+    assert_match /Failed to fetch_remote_dir.*/, exception.message, "#fetch_http_dir raised an unexpected RuntimeError"
```

Comment:
I believe `assert_raises` can take additional parameters, like the expected exception message.
So we could probably collapse these two assert calls into just the `assert_raises`